File: src/main/java/com/randomperson22/piggymodtest/network/PacketJumpscare.java
package com.randomperson22.piggymodtest.network;

import com.randomperson22.piggymodtest.events.Jumpscare_Handler;
import io.netty.buffer.ByteBuf;
import net.minecraft.client.Minecraft;
import net.minecraft.client.entity.EntityPlayerSP;
import net.minecraft.entity.Entity;
import net.minecraft.entity.EntityLivingBase;
import net.minecraft.network.PacketBuffer;
import net.minecraftforge.fml.common.network.simpleimpl.IMessage;
import net.minecraftforge.fml.common.network.simpleimpl.IMessageHandler;
import net.minecraftforge.fml.common.network.simpleimpl.MessageContext;
import net.minecraftforge.fml.relauncher.Side;

public class PacketJumpscare implements IMessage {
    private int entityId;
    private int duration;

    public PacketJumpscare() {}

    public PacketJumpscare(int entityId, int duration) {
        this.entityId = entityId;
        this.duration = duration;
    }

    @Override
    public void fromBytes(ByteBuf buf) {
        PacketBuffer buffer = new PacketBuffer(buf);
        this.entityId = buffer.readInt();
        this.duration = buffer.readInt();
    }

    @Override
    public void toBytes(ByteBuf buf) {
        PacketBuffer buffer = new PacketBuffer(buf);
        buffer.writeInt(this.entityId);
        buffer.writeInt(this.duration);
    }

    public static class Handler implements IMessageHandler<PacketJumpscare, IMessage> {
        @Override
        public IMessage onMessage(PacketJumpscare message, MessageContext ctx) {
            if (ctx.side == Side.CLIENT) {
                Minecraft.getMinecraft().addScheduledTask(() -> {
                    Entity e = Minecraft.getMinecraft().world.getEntityByID(message.entityId);
                    if (e instanceof EntityLivingBase) {
                        EntityPlayerSP player = Minecraft.getMinecraft().player;
                        Jumpscare_Handler.triggerJumpscareClient((EntityLivingBase) e, player, message.duration);
                    }
                });
            }
            return null; // no response packet
        }
    }
}

File: src/main/kotlin/com/randomperson22/piggymodtest/network/PacketJumpscare.kt
package com.randomperson22.piggymodtest.network

import com.randomperson22.piggymodtest.events.JumpscareHandler
import io.netty.buffer.ByteBuf
import net.minecraft.client.Minecraft
import net.minecraft.entity.EntityLivingBase
import net.minecraftforge.fml.common.network.simpleimpl.IMessage
import net.minecraftforge.fml.common.network.simpleimpl.IMessageHandler
import net.minecraftforge.fml.common.network.simpleimpl.MessageContext

class PacketJumpscare() : IMessage {
    private var entityId: Int = 0
    private var duration: Int = 0

    constructor(entityId: Int, duration: Int) : this() {
        this.entityId = entityId
        this.duration = duration
    }

    override fun toBytes(buf: ByteBuf) {
        buf.writeInt(entityId)
        buf.writeInt(duration)
    }

    override fun fromBytes(buf: ByteBuf) {
        entityId = buf.readInt()
        duration = buf.readInt()
    }

    class Handler : IMessageHandler<PacketJumpscare, IMessage> {
        override fun onMessage(message: PacketJumpscare, ctx: MessageContext): IMessage? {
            Minecraft.getMinecraft().addScheduledTask {
                val entity = Minecraft.getMinecraft().world.getEntityByID(message.entityId)
                val player = Minecraft.getMinecraft().player
                if (entity is EntityLivingBase) {
                    // Trigger jumpscare only for the targeted player
                    JumpscareHandler.triggerJumpscareClient(entity, player, message.duration)
                }
            }
            return null
        }
    }
}
